app-connector-client/server/routes/services.js

```diff
@@ -119,34 +119,36 @@ function fillServiceMetadata(serviceMetadata, api, hostname) {
             specInJson = yaml.safeLoad(fs.readFileSync(api.specification, 'utf8'));
         }
         serviceMetadata.api.spec = specInJson
-        serviceMetadata.api.specificationUrl = serviceMetadata.api.targetUrl + (api.metadata ? api.metadata : METADATA);
-        if (api.description) {
-            serviceMetadata.description = api.description;
-        } else if (specInJson.hasOwnProperty("info") && specInJson.info.hasOwnProperty("description")) {
-            serviceMetadata.description = specInJson.info.description;
-        }
-        else if (specInJson.hasOwnProperty("info") && specInJson.info.hasOwnProperty("title")) {
-            serviceMetadata.description = specInJson.info.title;
+
+        if (!api.description && specInJson.hasOwnProperty("info") && specInJson.info.hasOwnProperty("description")) {
+            api.description = specInJson.info.description
         }
-        else {
-            serviceMetadata.description = api.name;
+
+        if (!api.name && specInJson.hasOwnProperty("info") && specInJson.info.hasOwnProperty("title")) {
+            api.name = specInJson.info.title;

```

Comment:
shouldn't you set here serviceMetadata.api not api ?

Reply:
Am overriding the api name if not defined.
The name I use later down to set the description. I wanted to extract the description logic and use it for odata as well.